plop-templates/core/Component.js.hbs

```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import {
+  space,
+  themeGet
+} from 'styled-system'
+import {
+  mapProps,
+  applyVariations,
+  getPaletteColor
+} from './utils'
+
+const {{componentName}} = mapProps(({ align, ...props }) => ({
+  ...props
+}))(styled.{{htmlTag}}`
+  // Example: Use palette colors
+  color: ${getPaletteColor('base')};
+
+  // Advanced: Apply variation style along with custom styling from theme.componentStyles[component][variation]
+  ${applyVariations('{{componentName}}')}
+`)
+
+{{componentName}}.displayName = '{{componentName}}'
+
+{{componentName}}.propTypes = {
+}
+
+export default {{componentName}}
```

Comment:
Overall I wonder if the name `componentName` here and in other places (files) could be just `component` instead?

Reply:
I kinda wanted to make it obvious (even though it is an HBS template) that the thing we're inserting is a name